addons/web_editor/static/src/js/editor/add_snippet_dialog.js

```diff
@@ -311,6 +316,57 @@ export class AddSnippetDialog extends Component {
+                        // Auto-slide carousel snippets on hover
+                        const carouselSnippetEl = containerEl.querySelector(".carousel");
+                        if (carouselSnippetEl) {
+                            containerEl.addEventListener("mouseenter", function () {
+                                const carouselSnippetPreviewEl =
+                                    this.querySelector("[data-bs-ride]");
+                                carouselSnippetPreviewEl.dataset.bsInterval = 1000;
+                                const bsCarousel =
+                                    Carousel.getOrCreateInstance(carouselSnippetPreviewEl);
+                                bsCarousel.cycle();
+                            });
+                            containerEl.addEventListener("mouseleave", function () {
+                                const carouselSnippetPreviewEl =
+                                    this.querySelector("[data-bs-ride]");
+                                const bsCarousel =
+                                    Carousel.getOrCreateInstance(carouselSnippetPreviewEl);
+                                bsCarousel.pause();
+                                bsCarousel.to(0);
+                            });
+                        }
+                    } else if (containerEl.dataset.label === "Parallax") {
+                        // Manual parallax implementation is needed because
+                        // snippet previews are transformed to scale them down
+                        // and `background-attachment: fixed` doesn't work with
+                        // transformed elements.
+                        const parallaxSnippetEl = containerEl.querySelector(".parallax");
+                        const backgroundEl = parallaxSnippetEl.querySelector(".s_parallax_bg");
+                        const { body: iframeBody } = this.iframeDocument;
+
+                        const SCALE = 2.4; // Compensates for container overflow
+                        const PARALLAX_RATE = 0.7;
+                        const START_OFFSET = 400; // Delays effect until element is in view
+
+                        parallaxSnippetEl.style.overflow = "hidden";
+                        backgroundEl.style.transform = `scale(${SCALE})`;
+
+                        iframeBody.addEventListener("scroll", () => {
+                            const { top } = backgroundEl.getBoundingClientRect();
+                            const scrollOffset = window.innerHeight - START_OFFSET - top;
+                            if (scrollOffset > 0) {
+                                backgroundEl.style.transform = `translateY(${
+                                    scrollOffset * PARALLAX_RATE
+                                }px) scale(${SCALE})`;
+                            } else {
+                                backgroundEl.style.transform = `translateY(scale(${SCALE})`;
+                            }
+                        });
```

Comment:
Instead of triggering the handler fuction everytime we scroll on iframe, i would have done by adding an observer to it when its in the viewport then bind the scroll event to it and remove it when its not.